### PR TITLE
Fix invalid cypher when aggregating within nested relationships or connections

### LINKED ISSUE #2803: Invalid cypher when filtering by an aggregate in double nested relationships/connections
**Type definitions**

```graphql
type Movie {
    actors: [Actor!]! @relationship(type: "HAS_ACTOR", direction: OUT)
    released: Int!
}

type Actor {
    movies: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT)
    name: String
}
```

**Query**

```graphql
query {
    actors(where: { movies_SOME: { actors: { moviesAggregate: { count_GT: 0 } } } }) {
        name
    }
}
```

**Cypher**
```
MATCH (this:`Actor`)
OPTIONAL MATCH (this0)-[:HAS_ACTOR]->(this1:`Actor`) # This OPTIONAL MATCH should match all the way from this to this1 so the aggregation subquery has the correct info
CALL {
    WITH this1
    MATCH (this1)-[this2:ACTED_IN]->(this3:`Movie`)
    RETURN count(this3) > $param0 AS var4
}
WITH this0, collect(var4) AS var4 # This WITH should use this not this0
WITH *
WHERE EXISTS {
    MATCH (this)-[:ACTED_IN]->(this0:`Movie`)
    WHERE EXISTS {
        MATCH (this0)-[:HAS_ACTOR]->(this1:`Actor`)
        WHERE true IN var4
    }
}
```

### PR DESCRIPTION
**Pull request by Liam-Doodson**
# Description

Fixes invalid cypher when using aggregation filters within nested relationship/connection filters.

This issue arrises because a `CALL (subquery)` cannot be used within a `WHERE` clause. This means that aggregation subqueries cannot be used inside the existing relationship/connection filters which make use of `EXISTS`/`size()`/`single()`.

Therefore, if an aggregation filter is used, a `CALL (subquery)` is used to count the number of relationships matching any of the nested predicates. A `count > 0` indicates SOME, `count = 1` indicates SINGLE, etc...

## Complexity

Complexity: High

# Issue

Closes #2803

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2810.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2810.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 47 | 65 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 144 | 229 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 89 | 139 | 2174 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 78 | 120 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 89 | 162 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 5665 | 5916 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 111 | 139 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 375 | 384 | 3003 |
|  connections.Connection | 13042 | 13042 | 171 | 181 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 135 | 231 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 68 | 115 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2277 | 2277 | 298 | 447 | 1040 |
|  create.SimpleMutation | 6 | 6 | 34 | 43 | 1 |
|  cypher-directive.TopLevelCypherDirective | 99525 | 99525 | 863 | 878 | 10168 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 127992 | 127994 | 364 | 358 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 358 | 318 | 1040 |
|  delete.NestedDeleteInUpdate | 16779 | 16779 | 232 | 258 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 173 | 221 | 2174 |
|  query.QueryWhere | 8656 | 8659 | 42 | 68 | 2160 |
|  query.SimpleQueryWithNestedWhere | 8827 | 8830 | 62 | 92 | 2160 |
|  query.Nested | 10084290 | 10084290 | 11996 | 10862 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 170 | 258 | 4511 |
|  query.OrFilterOnRelationships | 36278 | 35775 | 215 | 464 | 1781 |
|  query.OrFilterOnRelationshipsAndNested | 30755 | 29456 | 239 | 641 | 1781 |
|  query.QueryWithNestedIn | 12227 | 12143 | 66 | 179 | 1177 |
|  query.NestedConnectionWhere | 8794 | 8794 | 77 | 180 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9054 | 8898 | 92 | 151 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 157 | 226 | 1134 |
|  query.DeeplyNestedWithRelationshipSingleFilters | 15438 | 15438 | 121 | 162 | 2174 |
|  query.Fulltext | 96 | 96 | 34 | 43 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 51 | 61 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 74 | 94 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1451 | 1457 | 269 | 504 | 1040 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 115 | 132 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 191 | 259 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 337 | 284 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 78 | 75 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 80 | 92 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 337 | 434 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 380 | 389 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 59 | 62 | 1040 |

</details>
        
Old Schema Generation: 42.608s
Schema Generation: 42.162s

File: packages/graphql/tests/integration/relationship-properties/connect-overwrite.int.test.ts [hunk 1]
```diff
@@ -54,8 +54,8 @@ describe("Relationship properties - connect with and without `overwrite` argumen
         let screenTimeUpdate: number;
 
         beforeEach(async () => {
-            typeActor = generateUniqueType("Actor");
-            typeMovie = generateUniqueType("Movie");
+            typeActor = new UniqueType("Actor");
+            typeMovie = new UniqueType("Movie");
```
Comment (angrykoala) [inline packages/graphql/tests/integration/relationship-properties/connect-overwrite.int.test.ts:58]: :clap:

**Review by angrykoala — APPROVED** [review]

**Pull request merged**